In [ ]:
!pip install pyspark


In [2]:
from pyspark.sql import SparkSession

# Start Spark
spark = SparkSession.builder.appName("BigDataAnalysis").getOrCreate()

# Load your_file.csv
df = spark.read.csv("/content/your_file.csv", header=True, inferSchema=True)

# Show first 5 rows
df.show(5)


+------+---+---+-------+-------+-------+----+----+-------+--------+------+--------+----------+---------+--------+---------+------+----+----------+-------+------+--------+--------+------+--------+-----+----+----+------+--------+------+
|school|sex|age|address|famsize|Pstatus|Medu|Fedu|   Mjob|    Fjob|reason|guardian|traveltime|studytime|failures|schoolsup|famsup|paid|activities|nursery|higher|internet|romantic|famrel|freetime|goout|Dalc|Walc|health|absences|passed|
+------+---+---+-------+-------+-------+----+----+-------+--------+------+--------+----------+---------+--------+---------+------+----+----------+-------+------+--------+--------+------+--------+-----+----+----+------+--------+------+
|    GP|  F| 18|      U|    GT3|      A|   4|   4|at_home| teacher|course|  mother|         2|        2|       0|      yes|    no|  no|        no|    yes|   yes|      no|      no|     4|       3|    4|   1|   1|     3|       6|    no|
|    GP|  F| 17|      U|    GT3|      T|   1|   1|at_home|  

In [3]:
# Drop rows with missing data
df = df.dropna()

# Count total rows after cleaning
print("Total rows after cleaning:", df.count())

# View column names to choose one for analysis
print("Columns:", df.columns)


Total rows after cleaning: 395
Columns: ['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu', 'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime', 'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery', 'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc', 'Walc', 'health', 'absences', 'passed']


In [4]:
df.groupBy("Mjob").count().orderBy("count", ascending=False).show(5)


+--------+-----+
|    Mjob|count|
+--------+-----+
|   other|  141|
|services|  103|
| at_home|   59|
| teacher|   58|
|  health|   34|
+--------+-----+



### 📈 Insights:
- Total rows after cleaning: 395
- Most common mother’s job is shown above.
- PySpark was used to handle and analyze student data efficiently.
